src/pywrapper.cpp

```diff
@@ -5,9 +5,21 @@
 
 #include "pywrapper.h"
 
-extern "C" {
 #include <Python.h>
-};
+
+// Inform user if a rather obscure python version is used
+#if PY_MAJOR_VERSION != 3
+#if PY_MAJOR_VERSION != 2
+#warning "Based on preprocessor macro PY_MAJOR_VERSION your not using python"
+#warning " with version 2 or 3. Inconsisties are expected!"
+#endif /* PY_MAJOR_VERSION != 2 */
+#endif /* PY_MAJOR_VERSION != 3 */
+
+// Define helper marco definition: python2 compiled for?
+#if PY_MAJOR_VERSION <= 2
+#define PYDEV_PYCAPI_COMPAT_VERSION2 1
+#endif /* PY_MAJOR_VERSION >= 3 */
+
```

Comment:
I would prefer to use existing Python macros (PY_MAJOR_VERSION) and not introduce new ones. We can deal with future distinctions later.

Reply:
I'll follow your preference.

I'd prefer to reduce the number of conditional compilations for handling strings, ints etc ...
I'll come back with a proposal.